### ISSUE TITLE
PsychoPy 2022.1.1 crashes starting the experiment Mac Monterey

### ISSUE THREAD

**Issue by karamellibisey** [psychopy/psychopy#4699]
Hello,

I've seen issues like this one being addressed before but sadly I couldn't find any solutions for it. 

Whenever I try to run an experiment, even with something as simple as a text component, I get the gray screen and then it crashes. 

After the crash, I get the following: 
<img width="604" alt="Screen Shot 2022-03-16 at 12 34 37" src="https://user-images.githubusercontent.com/101705600/158560472-6dee961b-6e95-4f10-9646-bd8ed4bc50eb.png">
<img width="630" alt="Screen Shot 2022-03-16 at 12 34 09" src="https://user-images.githubusercontent.com/101705600/158560692-46e1f977-bcaa-497d-9c9e-d26d7b175a6b.png">

I tried installing the old versions and allowing Psychopy to monitor input from my keyboard thing. 
Nothing seems to work. 

I'd appreciate the help :')

**Comment by kerenemily** [comment]
Also having this issue, program crashes with something as simple as a text object and no fixes I've seen before (i.e such as changing the project settings from the ioHub to another one, seem to help)...any help appreciated

**Comment by waltervanheuven** [comment]
Same issue here on an M1 MacBook Pro running Monterey 12.3. PsychoPy 2022.1.1 crashes with exit code -4.  I removed the `~/.psychopy3` folder and reinstalled PsychoPy 2021.2.3 and that works.

**Comment by peircej** [comment]
I think I've tracked down an issue on 2022.1.1 with the library gevent, which iohub relies on, and that certainly gives this symptom. I'm working on a fix (basically, hoping that updating the gevent lib is enough to fix it!)

**Comment by Wonbok83** [comment]
I download v2022.1.2 and Monterey 12.2.1, but I still have same problems ApplePersistentIgnoeState and couldn't connect to psychopy.org.
<img width="625" alt="Screen Shot 2022-03-28 at 11 45 05" src="https://user-images.githubusercontent.com/35937992/160436409-f50d0c9f-e286-4ed4-8f65-cc6eac39effe.png">

**Comment by KWik** [comment]
Since I happen to have both M1 and Intel Mac and Windows I checked and found this problem only occurs on my M1 MacBook Pro 14" and not on my MacBookPro13,2 (2016). Fresh install of Monterey 12.3.1 and PsychoPy 2022.1.2. On the old MacBook I get the old ioHub startup failed, but exit code 1, so everything is "fine". On windows 10 the single trial "Hello World" text runs as expected. 

Then I did some basic error checking, like determining if the screen setup was valid in the experiment Properties, with the goal of eliminating the Retina as a source of errors. Toying round with the resolution did not change the outcome, but it did eliminate some warnings.

Finally editing the experiment properties, I changed Input from ioHub to PsychToolbox (changing the exit code from -4 to -11) and finally to Pyglet. 

With Pyglet as Keyboard Backend the experiment displays as expected on both MacBook Pro 13" and on MacBook Pro 14" M1.

**Comment by KWik** [comment]
I was able to get this working by making a virtual environment for Python 3.8.10 and installing a fresh copy of gevent with pip install gevent==20.9.0 and replacing the one in the PsychoPy install with this. That only opens up a second error though, that one in

/Applications/PsychoPy.app/Contents/Resources/lib/python3.8/psychopy/iohub/client/keyboard.py

This can be fixed by adding a try on line 307:
```
                try:
                    key_events = Keyboard._iohubKeyboard.getReleases(keys=watchForKeys, clear=clear)
                except AttributeError as e:
                    key_events = []
```

On the Retina Macs you should probably also make sure that the monitor specified in the settings is _half_ the resolution available in system_profiler SPDisplaysDataType

EDIT: This was a short lived victory though, as the next time I ran an experiment it gave me a spinning beach ball and not much more.

**Comment by frdarya** [comment]
Same issue also on Big Sur (M1 Mac mini), downgrading to 2021.2.3 works

**Comment by kcm1231** [comment]
Hi there, any resolution on this issue? I'm having the same problem with PsychoPy v2022.1.2 and Mac Monterey (12.2). I tried to run the stroop demo and a few other simple experiments. The dialog box for participant and session number pops up, but as soon as I click 'ok' it crashes. The only error I'm getting is:

2022-04-12 09:39:27.747 python[45095:2359064] Warning: Expected min height of view: (<NSPopoverTouchBarItemButton: 0x7fbf3a5fbec0>) to be less than or equal to 30 but got a height of 32.000000. This error will be logged once per view in violation.
################ Experiment ended with exit code -4 [pid:45095]

**Comment by KWik** [comment]
Problem persists on PsychoPy v2022.1.3, Monterey 12.3
      Model Identifier: MacBookPro18,3
      Chip: Apple M1 Pro

**Comment by m-macaskill** [comment]
There may be several underlying causes for the crashes reported above, but the current situation for Mac users with up-to-date macOS and standalone PsychoPy is that _no experiment with default settings will run using Builder._ As the error message is not informative, users don't have any trouble-shooting path to follow. Unless they find the relevant posts on the forum or on Github, Builder is currently non-functional for them.

An immediate solution would be a bug-fix release that simply switches the default keyboard backend from `ioHub` to `PsychToolbox`.  Removing that universal source of errors would then allow any others to be dealt with more easily, as some of the posts above seem to indicate.

**Comment by peircej** [comment]
My apologies. I had tracked this down to a broken version of gevent and updated it (I thought) but it seems that hadn't made it into the released version. I've just compiled a patch release 2022.1.4 and confirmed it is working on my Big Sur installation. I just want to update to Monterey and check it still works there and should have that available for download tomorrow

**Comment by peircej** [comment]
OK, I believe this is now fixed in 2022.1.4 available for download at [PsychoPy releases](https://github.com/psychopy/psychopy/releases). There was, as I thought above, an issue withthe `gevent` module (required by iohub) which was crashing so aggressively it didn't give any error message. I've replaced that and all appears to be working fine now, but please do test that this has solved all the problems at your end. Thanks and apologies again

**Comment by junfang219** [comment]
I also ran into this issue. Psychopy 2023.2.3 and MacOS Sonoma, M1 Mac. The issue started happening today without me upgrading any libraries or OS versions
